### PR TITLE
Automate Kedro-viz Release process

### LINKED ISSUE #1477: Automate the release process
## Description

Our [release process](https://github.com/kedro-org/kedro-viz/blob/main/CONTRIBUTING.md#release-process) is manual. It would be better and probably safer if it was automated.

This ticket should automate the Kedro-Viz release process.

## Context

The Kedro framework release process is automated. It uses [some](https://github.com/kedro-org/kedro/blob/main/tools/circleci/github_scripts/release.sh) [scripts](https://github.com/kedro-org/kedro/blob/main/tools/circleci/circle-release.sh) (maybe more than only these two) to do the job. Let's use this as a template.

## Possible Implementation

See the context section above. I'd suggest forking the repo and doing the experimentation and implementation there first, then bring the working prototype over to this repo.

## Possible Alternatives

Try and use GitHub actions instead of `.sh` files?

## Checklist

- [x] Include labels so that we can categorise your feature request

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #1477 

## Development notes

Automated a part of the [release](https://github.com/kedro-org/kedro-viz/blob/main/CONTRIBUTING.md#release-process) process -

> - Once the release commit is in main, trigger the release by:
> - Create a Release through [Github UI](https://github.com/kedro-org/kedro-viz/releases/new).
> - Input the version as the tag version with Target set to main
> - Paste the release note in the text box.
> - Publish the release.

Tested Releases -
https://github.com/ravi-kumar-pilla/kedro-viz-forked/releases

image

## QA notes

**After automation, once the release commit is pushed to main -**

- A github action workflow (check-release) gets triggered which uses the python scripts check_version and extract_release_notes
- Generates a release tag and release notes based on RELEASE.md file

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1490"></a>

### CODE REVIEW

**Review by NeroOkwa — APPROVED** [review]

**Comment by tynandebold** [comment]
The code looks very good and it seems like it worked perfectly well in your forked repo. That's great!

I think we need some additional documentation about this. It isn't clear to me where the release version is coming from. I'm guessing it pulls from the `RELEASE.md` file, right? If so, we should add explicit wording to the `CONTRIBUTING.md` file to make that clear. Also, if we have something at the top of that file for the next upcoming release like the Framework team does, will that break anything?

Also, does the release branch name matter? Or can that be anything?

My suggestion is to think about the documentation part of this as if you didn't write this PR. For example, if you were a brand new member of the Viz team tasked with releasing a new version, would you be able to do it with the current documentation?

**Review by tynandebold — COMMENTED** [review]
The code looks very good and it seems like it worked perfectly well in your forked repo. That's great!

I think we need some additional documentation about this. It isn't clear to me where the release version is coming from. I'm guessing it pulls from the RELEASE.md file, right? If so, we should add explicit wording to the CONTRIBUTING.md file to make that clear. Also, if we have something at the top of that file for the next upcoming release like the Framework team does, will that break anything?

Also, does the release branch name matter? Or can that be anything?

My suggestion is to think about the documentation part of this as if you didn't write this PR. For example, if you were a brand new member of the Viz team tasked with releasing a new version, would you be able to do it with the current documentation?

File: tools/github_actions/check_version.py
Comment (tynandebold) [inline tools/github_actions/check_version.py:1]: Should this file go in the `.github` folder instead?

File: tools/github_actions/extract_release_notes.py
Comment (tynandebold) [inline tools/github_actions/extract_release_notes.py:1]: Same with this file?

File: tools/github_actions/check_version.py
Comment (ravi-kumar-pilla) [reply tools/github_actions/check_version.py:1]: I followed how kedro-plugins github actions were structured by Ankita here -  https://github.com/kedro-org/kedro-plugins/blob/main/.github/workflows/check-release.yml

Also, we have versioning.py logic inside tools which is related to make version command. So, I thought this to be a good place.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -323,7 +323,10 @@ If the lower-bound e2e test fails, it indicates that some dependencies may not w
 
 ## Release process
 
-- Update [RELEASE.md](./RELEASE.md) with the latest release note.
+- Create a branch for release. The branch name should follow the pattern `release/* (e.g. release/6.4.0)`
+- Update [RELEASE.md](./RELEASE.md) with the latest release note. 
+
+> _Notes_: The heading for the release should follow the pattern `# Release <release-number> (e.g. # Release 6.4.0)`
```
Comment (tynandebold) [inline CONTRIBUTING.md:329]:
```suggestion
> _Note_: The heading for the release note should follow the pattern `# Release <release-number> (e.g. # Release 6.4.0)`
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -333,13 +336,9 @@ make version VERSION=<version-to-release>
 > _Notes_: Kedro-Viz uses [Semantic Versioning](https://semver.org/) as the versioning scheme.
 
 - Commit, raise a PR and get it merged into `main`.
+- Once the release commit is in `main`, a github action workflow gets triggered which generates a release tag and notes based on [RELEASE.md](./RELEASE.md) to publish.
```
Comment (tynandebold) [inline CONTRIBUTING.md:339]:
```suggestion
- Once the release commit is in `main`, a GitHub action gets triggered which automatically publishes a release with a version tag and release notes based on [RELEASE.md](./RELEASE.md).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -333,13 +336,9 @@ make version VERSION=<version-to-release>
...
 
-- Once the release commit is in `main`, trigger the release by:
-
-  - Create a Release through [Github UI](https://github.com/kedro-org/kedro-viz/releases/new).
-  - Input the version as the tag version with Target set to `main`
-  - Paste the release note in the text box.
-  - Publish the release.
+> _Notes_: The release tag corresponds to `__version__` from `package/kedro_viz/__init__.py` and the release notes is the content under the heading `# Release <release-number> (e.g. # Release 6.4.0)` in [RELEASE.md](./RELEASE.md).
```
Comment (tynandebold) [inline CONTRIBUTING.md:341]:
```suggestion
> _Note_: The release tag corresponds to `__version__` from `package/kedro_viz/__init__.py` and the release notes is the content under the heading `# Release <release-number> (e.g. # Release 6.4.0)` in [RELEASE.md](./RELEASE.md).
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -333,13 +336,9 @@ make version VERSION=<version-to-release>
...
 
 - Once the release tag is published on Github, a CircleCI job will be triggered to push it to npm and PyPI accordingly.
```
Comment (tynandebold) [inline CONTRIBUTING.md:343]:
```suggestion
- Once the release tag is published on GitHub, a CircleCI job will be triggered to push it to npm and PyPI accordingly.
```

**Review by tynandebold — APPROVED** [review]
Looks great. Thanks for adding the additional checks.

Approving with some small nit comments regarding language and style.

File: tools/github_actions/extract_release_notes.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    for i, line in enumerate(lines):
+        if line.startswith("# "):
+            current_heading = line.strip("#").replace(":", "").strip()
+            if current_heading == heading:
+                start_line = i
+            elif start_line is not None:
+                end_line = i
+                break
+
+    if start_line is not None:
```
Comment (rashidakanchwala) [inline tools/github_actions/extract_release_notes.py:19]:
```suggestion
    ```suggestion
    if start_line is None:
        return None

    end_line = end_line or len(lines)
    section = "".join(lines[start_line + 1 : end_line]).strip()
    return section
```
```

**Review by rashidakanchwala — APPROVED** [review]
This is great ! thanks @ravi-kumar-pilla

**Pull request merged**